pkg/templatelookup/regular_test.go

```diff
@@ -893,18 +835,23 @@ func TestTemplateLookup_GetRegularTemplates_WhenModuleTemplateNotFound(t *testin
 			want: templatelookup.ModuleTemplatesByModuleName{
 				testModule.Name: &templatelookup.ModuleTemplateInfo{
 					DesiredChannel: testModule.Channel,
-					Err:            templatelookup.ErrNoTemplatesInListResult,
+					Err:            moduletemplateinfolookup.ErrNoTemplatesInListResult,
 				},
 			},
 		},
 	}
 	for _, testCase := range tests {
 		t.Run(testCase.name, func(t *testing.T) {
 			givenTemplateList := &v1beta2.ModuleTemplateList{}
-			lookup := templatelookup.NewTemplateLookup(NewFakeModuleTemplateReader(*givenTemplateList,
-				v1beta2.ModuleReleaseMetaList{}),
+			reader := NewFakeModuleTemplateReader(*givenTemplateList,
+				v1beta2.ModuleReleaseMetaList{})
+			lookup := templatelookup.NewTemplateLookup(reader,
 				provider.NewCachedDescriptorProvider(),
-				maintenanceWindowStub{})
+				moduletemplateinfolookup.NewAggregatedModuleTemplateInfoLookupStrategy([]moduletemplateinfolookup.ModuleTemplateInfoLookupStrategy{
```

Comment:
Since these dependencies are now abstracted via interface, the unit tests of `TemplateLookup` should work on mock implementations.